pydantic/types.py

```diff
@@ -515,7 +515,7 @@ def __get_validators__(cls) -> 'CallableGenerator':
         yield cls.validate
 
     @classmethod
-    def validate(cls, v: str) -> Any:
+    def validate(cls, v: Union[str]) -> Any:
```

Comment:
If you use just `str` as the type annotation, Cython compiles the function in such a way that a non-str will cause a TypeError. This caused some tests to fail. Changing to Union[str] prevents Cython from performing this optimization, but I don't think it has any implications for mypy.